src/main/java/com/sda/vehicles/VehicleApp.java

```diff
@@ -3,16 +3,39 @@
 public class VehicleApp {
     public static void main(String[] args) {
         Vehicle vehicle1 = new Vehicle();
+        vehicle1.run(130);
+        System.out.println(vehicle1);
 
-        vehicle1.run(100.0);
-        vehicle1 = new Car();
-        vehicle1.run(50);
+        System.out.println();
 
+        Car car = new Car();
+        car.addPassenger();
+        car.run(112.5);
+        System.out.println(car);
         System.out.println();
+
         System.out.println();
 
-        Car car = new Car();
+        Vehicle truck = new Truck();
+        truck.run(120);
+        System.out.println(truck);
+        ((Truck) truck).addPackages();
```

Comment:
W takiej sytuacji, w której chciałbyś użyć metod konkretnej subklasy, lepiej jest od razu stworzyć zmienną tej subklasy i uniknąć rzutowania. Czyli zamiast `Vehicle truck = new Truck()` -> `Truck truck = new Truck()`.

To nie błąd - polimorfizmu używamy gdy zależy nam na zmiennej, która jest uniwersalna, czyli gdyby nas nie interesowało jaką dokładnie implementację `Vehicle` ma pod sobą. Tutaj z kolei ważne jest dla nas by zmienna była typu `Truck`, bo chcemy skorzystać z metod obecnych `Truck` - załadować go paczkami. :) 

No i ogólnie działa zasada - jak widzę rzutowanie w code review to dwa razy zastanawiam się czy na pewno jest ok. :)